### PR TITLE
improve preview reliability

### PR DESCRIPTION
**Pull request by FredKSchott**
## Changes

- Improves some preview logic for performance and bug fixes
- Fixes flakey CI tests where `close()` would resolve before the server actually closed

## Testing

- Covered by CI

## Docs

- N/A

### CODE REVIEW

File: packages/astro/src/core/preview/index.ts [hunk 1]
```diff
@@ -30,7 +30,10 @@ export default async function preview(config: AstroConfig, { logging }: PreviewO
 	const trailingSlash = config.devOptions.trailingSlash
 	/** Base request URL. */
 	let baseURL = new URL(config.buildOptions.site || '/', defaultOrigin);
-
+	const staticFileServer = sirv(fileURLToPath(config.dist), {
```
Comment (FredKSchott) [inline packages/astro/src/core/preview/index.ts:33]: Fix: we shouldn't create a new instance on every request. This should speed up performance considerably if a site requests many assets per page.

File: packages/astro/src/core/preview/index.ts [hunk 2]
```diff
@@ -48,28 +51,22 @@ export default async function preview(config: AstroConfig, { logging }: PreviewO
...
 			res.statusCode = 404;
 			res.end(notFoundTemplate(pathname, message));
 		};
 
-		switch(true) {
+		switch (true) {
 			case hasTrailingSlash && trailingSlash == 'never' && !isRoot:
-				err('Prohibited trailing slash');
-				break;
-			case !hasTrailingSlash && trailingSlash == 'always' && !isRoot:
```
Comment (FredKSchott) [inline packages/astro/src/core/preview/index.ts:60]: I think this was a bug, where every URL required a trailing slash which would have made URLs like `/favicon.png` fail

File: packages/astro/src/core/preview/index.ts [hunk 3]
```diff
@@ -122,7 +119,9 @@ export default async function preview(config: AstroConfig, { logging }: PreviewO
 		port,
 		server: httpServer!,
 		stop: async () => {
-			httpServer.close();
+			await new Promise((resolve, reject) => {
```
Comment (FredKSchott) [inline packages/astro/src/core/preview/index.ts:122]: Flakey CI fix #1

File: packages/astro/test/0-css.test.js [hunk 4]
```diff
@@ -25,7 +25,7 @@ describe('CSS', function () {
 
 	// test HTML and CSS contents for accuracy
 	describe('build', () => {
-		this.timeout(30000); // test needs a little more time in CI
+		this.timeout(45000); // test needs a little more time in CI
```
Comment (FredKSchott) [inline packages/astro/test/0-css.test.js:28]: Flakey CI fix #2

**Review by natemoo-re — APPROVED** [review]
Nice catches! Approved.

**Pull request merged**